### PR TITLE
gh-108747: Add unit tests for site.{usercustomize,sitecustomize} hooks

### PR DESCRIPTION
**Pull request by csm10495**
Add missing unit tests

* Issue: gh-108747

### CODE REVIEW

**Comment by csm10495** [comment]
@hauntsaninja, I think you were the last person to edit site.py. Any chance I can have a review? Thanks!

**Comment by csm10495** [comment]
Friendly ping @hauntsaninja. Thanks.

File: Lib/test/test_site.py [hunk 1]
```diff
@@ -465,6 +465,59 @@ def test_sitecustomize_executed(self):
...
+                os.makedirs(base_path, exist_ok=True)
+            except PermissionError:
+                # Can't modify system site packages depending on the system configuration
+                continue
+
+            customize_path = os.path.join(base_path, f'{module_name}.py')
+            if os.path.exists(customize_path):
+                # backup old sitecustomize.py
+                oldcustomize_path = customize_path + '.old'
+                if os.path.exists(oldcustomize_path):
```
Comment (hauntsaninja) [inline Lib/test/test_site.py]: I don't know that I love this, could we instead write a sitecustomize into a temp dir, and then put that dir into `PYTHONPATH` env var of the subprocess?

**Review by hauntsaninja — COMMENTED** [review]
Thanks for adding a test! Not sure that I'm the right person to review, but left a comment :-)

**Comment by csm10495** [comment]
Updated. @hauntsaninja if you have someone else in mind, feel free to add them. Thanks still.

**Review by hauntsaninja — APPROVED** [review]
Thank you!

**Pull request merged**